### PR TITLE
feat: added the logger to REST API requests in KafkaConnectApiImpl

### PR DESCRIPTION
**Pull request by omjikush09**
### Type of change
- Enhancement / new feature

### Description
 Pull request for  [issue](https://github.com/strimzi/strimzi-kafka-operator/issues/8055). Added logger for REST API at debug level. 
### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Make sure all tests pass
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java [hunk 1]
```diff
@@ -285,6 +286,7 @@ public Future<Void> resume(String host, int port, String connectorName) {
     }
 
     private Future<Void> pauseResume(String host, int port, String path) {
+
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java:289]: I don't think we need this empty line here. Should we add the log message instead here?

**Review by scholzj — COMMENTED** [review]
Thanks for the Pr. It looks mostly good. I left one comment ... also, it seems like the logging is missing also in the `list(...)` method?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java (hunk 1 above)
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java:289]: Agree

**Review by ppatierno — APPROVED** [review]
LGTM. Unless the comment from Jakub.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java (hunk 1 above)
Comment (omjikush09) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java:289]: Hi, will add a logger there also ( in `pause` and `resume` ). But Logger requires `reconciliation` as a parameter so need to change the signature of pause and resume as instructed by @kyguy. If you have anything to suggest please let me know.
Will push the code soon.

**Review by scholzj — APPROVED** [review]

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java (hunk 1 above)
Comment (kyguy) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java]: @omjikush09 Looking solid, great work! Just address the checkstyle errors and make sure the unit tests are passing

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR.

**Pull request merged**